Dockerfile

```diff
@@ -1,20 +1,22 @@
 FROM rust:1-slim-bookworm AS build
-ARG DEBIAN_FRONTEND=noninteractive
 
-ADD . /app
 WORKDIR /app
-RUN apt-get update \
-  && apt-get install -y pkg-config libssl-dev cmake \
-  && cargo build --release
+COPY . .
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends pkg-config libssl-dev cmake && \
+    cargo build --release
 
 FROM debian:bookworm-slim
-ARG DEBIAN_FRONTEND=noninteractive
 
-RUN adduser --uid 1001 --group --no-create-home --home /app lava-gitlab-runner
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends libssl3 ca-certificates && \
+    rm -rf /var/lib/apt/lists/* && \
+    groupadd -g 1001 lava-gitlab-runner && \
+    useradd -u 1001 -g lava-gitlab-runner -d /app -M lava-gitlab-runner
```

Comment:
Keep one logical change per RUN actions ; adduser and installing libssl are two very orthogonal things